### PR TITLE
Support unification for `Micro` module

### PR DESCRIPTION
**Pull request by KhraksMamtsov**
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related

- Related Issue #
- Closes #

### CODE REVIEW

File: packages/effect/src/Micro.ts [hunk 1]
```diff
@@ -64,9 +65,27 @@ export type runSymbol = typeof runSymbol
...
   readonly [runSymbol]: (env: Env<any>, onResult: (result: Result<A, E>) => void) => void
+  [Unify.typeSymbol]?: unknown
+  [Unify.unifySymbol]?: MicroUnify<this>
+  [Unify.ignoreSymbol]?: MicroUnifyIgnore
   [Symbol.iterator](): MicroIterator<Micro<A, E, R>>
 }
 
+/**
+ * @category models
+ * @since 2.0.0
```
Comment (tim-smart) [inline packages/effect/src/Micro.ts:76]:
```suggestion
 * @since 3.4.3
```

File: packages/effect/src/Micro.ts [hunk 2]
```diff
@@ -64,9 +65,27 @@ export type runSymbol = typeof runSymbol
...
+ */
+export interface MicroUnify<A extends { [Unify.typeSymbol]?: any }> extends EffectUnify<A> {
+  Micro?: () => A[Unify.typeSymbol] extends Micro<infer A0, infer E0, infer R0> | infer _ ? Micro<A0, E0, R0> : never
+}
+
+/**
+ * @category models
+ * @since 2.0.0
```
Comment (tim-smart) [inline packages/effect/src/Micro.ts:84]:
```suggestion
 * @since 3.4.3
```

File: .changeset/rare-cameras-explain.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"effect": minor
```
Comment (tim-smart) [inline .changeset/rare-cameras-explain.md:2]:
```suggestion
"effect": patch
```

**Review by tim-smart — CHANGES_REQUESTED** [review]

**Comment by KhraksMamtsov** [comment]
Don't know how to change target brunch(
Bot has changed it from the `main`

**Comment by mikearnaldi** [comment]
> Don't know how to change target brunch( Bot has changed it from the `main`

Done, you need to hit Edit on the PR title, it's an odd ux imho

**Review by tim-smart — APPROVED** [review]

**Pull request merged**